pkg/yurthub/pki/certmanager/certmanager.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2020 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package certmanager
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"fmt"
+	"net"
+
+	"github.com/openyurtio/openyurt/pkg/projectinfo"
+
+	certificates "k8s.io/api/certificates/v1beta1"
+	"k8s.io/client-go/kubernetes"
+	clicert "k8s.io/client-go/kubernetes/typed/certificates/v1beta1"
+	"k8s.io/client-go/util/certificate"
+	"k8s.io/klog/v2"
+)
+
+const (
+	YurtHubServerCSROrg = "system:masters"
+	YurtHubCSROrg       = "openyurt:yurthub"
+	YurtHubServerCSRCN  = "kube-apiserver-kubelet-client"
+)
+
+// NewYurtHubServerCertManager creates a certificate manager for
+// the yurthub-server
+func NewYurtHubServerCertManager(
+	clientset kubernetes.Interface,
+	certDir string) (certificate.Manager, error) {
+
+	klog.Infof("subject of yurthub server certificate")
+	return newCertManager(
+		clientset,
+		projectinfo.GetHubName(),
```

Comment:
make sure the name of yurthub server certificate is different from yurthub client certificate, so componentName should not use `projectinfo.GetHubName()`.  maybe ``projectinfo.GetHubName()-server` is more reasonable.